airbyte-webapp/src/core/request/AirbyteRequestService.ts

```diff
@@ -1,28 +1,53 @@
 import { NetworkError } from "core/request/NetworkError";
 import config from "config";
+import { VersionError } from "./VersionError";
 
 abstract class AirbyteRequestService {
   static rootUrl = config.apiUrl;
 
-  /** Perform network request*/
+  /** Perform network request */
   static async fetch(
     url: string,
     body?: Readonly<Record<string, unknown> | Array<unknown> | string>,
     options?: Partial<RequestInit>
   ): Promise<Response> {
-    const response = await fetch(url, {
+    const path = `${this.rootUrl}${url}`;
+    const response = await fetch(path, {
       method: "POST",
       body: body ? JSON.stringify(body) : undefined,
       ...options,
     });
 
+    return this.parseResponse(response);
+  }
+
+  /** Parses errors from server */
+  static async parseResponse<T>(response: Response): Promise<T> {
+    if (response.status === 204) {
+      return {} as T;
+    }
     if (response.status >= 200 && response.status < 300) {
-      return response;
-    } else {
-      const e = new NetworkError(response);
-      e.status = response.status;
-      throw e;
+      return response.status === 204 ? {} : await response.json();
+    }
+    let result: any;
+
+    // If some error returned in json, lets try to parse it
+    try {
+      result = await response.json();
+    } catch (e) {
+      //
+    }
+
+    if (result?.error) {
+      if (result.error.startsWith("Version mismatch between")) {
```

Comment:
I'm going to create an [issue](https://github.com/airbytehq/airbyte/issues/3666) for the API to standardize the schema of its error messages. I agree with your choice here for now, but would like to have things be a bit more structured in the future, but the API has to lead on this.

Reply:
Exactly! Totally forgot to ask also send some sort of id for errors. But we could update it later